Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/revert.py

```diff
@@ -227,10 +235,42 @@ def revert_commit(cls, args, repository, issue, commit_objects, commit_bugs, com
             return 1
         log.info('Reverted {}'.format(', '.join(commits)))
 
-        # This should occur only after a successful revert.
-        for c_bug in commit_bugs:
-            # TODO: relate revert bug
-            c_bug.open(why="{}, tracking revert in {}".format(revert_reason, issue.link))
+        return 0
+
+    @classmethod
+    def relate_issues(cls, args, repository, issue, commit_bugs, commit_radars, revert_reason):
```

Comment:
Looking through this function, the only bits that are unique to specific trackers are `cc_radar`, catching the `UnsuccessfulResponseException` and `blocks` vs `cause_of`.

In general, you basically have two buckets. The original issues (in all bug trackers) and then the revert issues (also in all bug trackers). You're trying to re-open all original issues and relate them to the paired revert issue in their specific bug tracker. I would like to see if we could use unified logic until the very last moment, and ask ourselves if there is any way to unify even that last moment (there may not be, I don't think `cc_radar` has an idiom that makes sense when invoked on a radar, for example)

Reply:
That's a great summary! I tried implementing it so we only need to condition on tracker type when relating as you mentioned. I checked and `cc_radar` will return `None` if called on a radar which is nice.